src/gateways/GoPayRecurrent.php

```diff
@@ -248,4 +250,15 @@ protected function getError(): ?array
         }
         return reset($this->response->getData()['errors']);
     }
+
+    protected function checkChargeStatus($payment, $resultCode)
+    {
+        if (!$this->isPendingState($this->response->getCode())) {
```

Comment:
This is only here because omnipay library "stops" only for `CREATED` state and not for the `PAYMENT_METHOD_CHOSEN` state. Correct?

https://github.com/tomaj/omnipay-gopay/blob/master/src/Message/RecurrenceRequest.php#L60

If so, shouldn't we also make a PR for the omnipay library to fix the issue? We could then get rid of this override of `checkChargeStatus`.

(note: it still feels weird, that backend payment should be in the `PAYMENT_METHOD_CHOSEN` state)

Reply:
Maybe we could also do an option for omnipay library not to wait on the `CREATED` state and return it immediately? That way the charging would be much faster and confirmations would be done through the notifications.

Reply:
Correct, the omnipay lib considers the payment as successful only in the CREATED state, otherwise thinks that it failed.. which is not true because of the PAYMENT_METHOD_CHOSEN state.. it would be great if we could update the omnipay library, but bileto/omnipay-gopay is not maintained, at least it looks like it's not.. the RecurrenceRequest is different from the fork..

Reply:
Ouch, I just realized the base omnipay package is not in the composer JSON. So you're using bileto's library instead of fork made by @tomaj ? All of our testing is done against https://github.com/tomaj/omnipay-gopay.

Reply:
btw I don't think that the status check should be done that way (while).. according to our logs, the payment status change takes around 20-30 seconds, that means 10-15 requests to gopay and a lot of waiting in the charge command..

Reply:
Yeah, that's probably correct too. How about this:

1. Let's make PR into tomaj's library with the changes:
  - `RecurrenceResponse::isSuccessful` remains the same, returning `true` only for `PAID` status
  - add `RecurrenceResponse::isPending` method which returns `true` for `CREATED` and `PAYMENT_METHOD_CHOSEN` status
  - get rid of the `while()` loop in the `RecurrenceRequest` to speed up charging and wait for the notification to confirm the payment.
2. We'll add tomaj's library into module's composer JSON. It should have been there from the beginning - that's what we're using for testing.

Would that make sense?

Reply:
That could work, but isSuccessful gets called in the GatewayAbstract class too I think. Could it make any problems there?

Reply:
Yeah, it could. We still will need to override the `checkChargeStatus` because of the "pending" state handling, so I will probably merge this as it is anyway. The changes in the omnipay library will allow us 1) to charge people faster and 2) let omnipay library to tell us "what is *pending* state".

Can you please test this with tomaj's library before I merge this? I'd like to include the missing dependency in `composer.json` file.

Reply:
To give everybody the full picture: After the offline discussion we'll make `mikoczy/omnipay-gopay` library the one that this module uses, since the implementation matches the expectations above and also adds some extra features.